src/backend/distributed/executor/multi_utility.c

```diff
@@ -280,10 +326,18 @@ multi_ProcessUtility(Node *parsetree,
 	{
 		if (IsA(parsetree, IndexStmt))
 		{
-			/* copy parse tree since we might scribble on it to fix the schema name */
-			parsetree = copyObject(parsetree);
+			/*
+			 * copy parse tree since we might scribble on it to fix the schema name
+			 * FIXME: PostgreSQL 10 changes caching behavior and breaks this approach,
+			 *        so for that version we do assign back the modified tree.
+			 */
```

Comment:
@marcocitus @jasonmp85 I'm not really ok with just regressing here.  The likelihood of tackling that anytime soon seems close to zero, and it'll get copied around making the problem harder and harder to fix.